crates/providers/kv-redis/src/bin/kvredis.rs

```diff
@@ -148,72 +150,96 @@ impl ProviderHandler for KvRedisProvider {
     #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor, key = %arg.key))]
-    async fn increment(&self, ctx: Context, arg: IncrementRequest) -> Result<i32, String> {
+    async fn increment(
+        &self,
+        ctx: Context,
+        arg: IncrementRequest,
+    ) -> ProviderInvocationResult<i32> {
         let mut cmd = redis::Cmd::incr(&arg.key, arg.value);
-        let val: i32 = self.exec(&ctx, &mut cmd).await?;
+        let val: i32 = self
+            .exec(&ctx, &mut cmd)
+            .await
+            .map_err(ProviderInvocationError::Provider)?;
         Ok(val)
     }
 
     /// Returns true if the store contains the key
     #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor, key = %arg.to_string()))]
-    async fn contains(&self, ctx: Context, arg: String) -> Result<bool, String> {
+    async fn contains(&self, ctx: Context, arg: String) -> ProviderInvocationResult<bool> {
         let mut cmd = redis::Cmd::exists(arg.to_string());
-        let val: bool = self.exec(&ctx, &mut cmd).await?;
+        let val: bool = self
+            .exec(&ctx, &mut cmd)
+            .await
+            .map_err(ProviderInvocationError::Provider)?;
         Ok(val)
     }
 
     /// Deletes a key, returning true if the key was deleted
     #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor, key = %arg.to_string()))]
-    async fn del(&self, ctx: Context, arg: String) -> Result<bool, String> {
+    async fn del(&self, ctx: Context, arg: String) -> ProviderInvocationResult<bool> {
         let mut cmd = redis::Cmd::del(arg.to_string());
-        let val: i32 = self.exec(&ctx, &mut cmd).await?;
+        let val: i32 = self
+            .exec(&ctx, &mut cmd)
+            .await
+            .map_err(ProviderInvocationError::Provider)?;
         Ok(val > 0)
     }
 
     /// Gets a value for a specified key. If the key exists,
     /// the return structure contains exists: true and the value,
     /// otherwise the return structure contains exists == false.
     #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor, key = %arg.to_string()))]
-    async fn get(&self, ctx: Context, arg: String) -> Result<GetResponse, String> {
+    async fn get(&self, ctx: Context, arg: String) -> ProviderInvocationResult<GetResponse> {
         let mut cmd = redis::Cmd::get(arg.to_string());
-        let val: Option<String> = self.exec(&ctx, &mut cmd).await?;
+        let val: Option<String> = self
+            .exec(&ctx, &mut cmd)
+            .await
+            .map_err(ProviderInvocationError::Provider)?;
+
         let resp = match val {
             Some(s) => GetResponse {
                 exists: true,
                 value: s,
             },
             None => GetResponse {
                 exists: false,
-                ..Default::default()
+                value: String::default(),
```

Comment:
Is there a reason for this refactor?

Reply:
As with just about all the refactors in this PR, there is a reason. 

The generated `GetResponse` struct does not implement `Default` -- so this code does not work. It worked with hand-coded structs but does not with the generated struct. The question of whether we should/*can* auto-derive `Default` is separate and should be handled outside this PR